app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+  belongs_to :user
+
+  def recent_5_comments
+    comment.order('created_at Desc').limit(5)
+  end
+
+  after_save :update_posts_counter
+
+  private
+
+  def update_posts_counter
+    user.increment!(:posts_counter)
+  end
```

Comment:
Great work here using [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database. 👏👏

Reply:
Hi @petudeveloper  I do not think I need to use class_name: 'User' as the user_id is the foreign key I am using and that will automatically match the User class